### ISSUE TITLE
`tf.reduce_min/max/mean` returns uninitialized value on tensors with zero-size dimension

### ISSUE THREAD

**Issue by bgshih** [tensorflow/tensorflow#3510]
I find that `tf.reduce_min/max/mean` returns uninitialized value (`nan` for example) on tensors with a zero-size dimension. This would cause problems, for example, when computing loss in an object detection problem, where an input image may contain no object. Please consider defining its behaviour on such inputs, e.g. outputing zeros, raising errors, etc.

### Environment info

  * Operating System: Ubuntu 14.04 LTS
  * CUDA 7.5, cuDNN 4.0
  * TF version: 0.9.0rc0, pip version (gpu, python 2.7)

### Steps to reproduce

    import tensorflow as tf
    with tf.Session():
      print(tf.reduce_max(tf.constant(-1.0, shape=[0, 10])).eval())

Output:

    3.40282e+38

Also works for `tf.reduce_min`, `tf.reduce_mean`, but not for `tf.reduce_sum`.

**Comment by michaelisard** [comment]
This is the expected behavior. When a reduction is performed over a tensor with zero elements and the result does not have zero elements, the result is filled with the reducer's identity function. For max that is 'lowest' for the type in question, which is what you should be seeing. (When I try this I see -3.4e+38 for max, and +3.4e+38 for min since min's identity is highest.)

Please reopen if you aren't seeing this behavior.

**Comment by bgshih** [comment]
@michaelisard I see. Thanks for explaining!
